python/src/iceberg/types.py

```diff
@@ -48,27 +46,69 @@ def __str__(self):
 
     @property
     def is_primitive(self) -> bool:
-        return self._is_primitive
+        return isinstance(self, PrimitiveType)
+
 
+class PrimitiveType(Type):
+    """Base class for all Iceberg Primitive Types"""
+
+    _implemented = {}  # type: ignore
+
+    def __new__(cls):
```

Comment:
What do you think the benefits of that might be?

Reply:
Possible reuse later.

Reply:
I'd probably continue to use Singleton so that you don't have to call `object.__new__` and can use `super().__new__` instead. That seems safer to me for some reason.